### DISCUSSION TITLE
NumberInput and Loader Type issue

### DISCUSSION THREAD

**Discussion by frec-hansen** [mantinedev/mantine#4970] [Q&A] [answered]
In V6, NumberInput `onChange` used to give a value of type `number | ""`. Now this have changed in V7 to be `number | string`. 
I couldn't find any documentation that suggests when it would return a `string` and when it would return a `number`. 
Is the use case still the same where if it only returns empty string `""` or a number?

Load allows custom loading according to [this](https://mantine.dev/core/loader/#custom-css-only-loader). However, when I use it in combination with the `LoaderLayout`, TypeScript complains about the key `custom`:
<img width="348" alt="Screenshot 2023-10-04 at 5 19 59 PM" src="https://github.com/mantinedev/mantine/assets/97467751/6c21d9fc-6312-41e9-9c26-f2923d2f4642">

**Comment by rtivital** [comment, marked as answer]
Starting from 7.0, NumberInput was migrated to react-number-format. It supports string representation of number. If you use numbers, then it will work the same way as in v6.

**Reply by frec-hansen** [reply]
Sorry I guess I am not understanding the use case correctly. It looks like `react-number-format` can [just return a numerical value](https://s-yadav.github.io/react-number-format/docs/props/#onvaluechange-values-sourceinfo--). Can we just take that numerical value and use it in the `onChange` so it always returns a `number` instead of `number | string`?

**Reply by rtivital** [reply]
No, the types of NumberInput props are static, you should add additional validation on your side if you need this feature
